Foundation/NSByteCountFormatter.swift

```diff
@@ -40,49 +40,130 @@ public enum NSByteCountFormatterCountStyle : Int {
 }
 
 public class NSByteCountFormatter : NSFormatter {
+    public override init() {
+        super.init()
+    }
+    
     public required init?(coder: NSCoder) {
         NSUnimplemented()
     }
     
     /* Shortcut for converting a byte count into a string without creating an NSByteCountFormatter and an NSNumber. If you need to specify options other than countStyle, create an instance of NSByteCountFormatter first.
     */
-    public class func stringFromByteCount(byteCount: Int64, countStyle: NSByteCountFormatterCountStyle) -> String { NSUnimplemented() }
-    
+    public class func stringFromByteCount(byteCount: Int64, countStyle: NSByteCountFormatterCountStyle) -> String {
+        let byteCountFormatter = NSByteCountFormatter()
+        byteCountFormatter.countStyle = countStyle
+        return byteCountFormatter.stringFromByteCount(byteCount)
+    }
+
     /* Convenience method on stringForObjectValue:. Convert a byte count into a string without creating an NSNumber.
     */
     public func stringFromByteCount(byteCount: Int64) -> String { NSUnimplemented() }
     
     /* Specify the units that can be used in the output. If NSByteCountFormatterUseDefault, uses platform-appropriate settings; otherwise will only use the specified units. This is the default value. Note that ZB and YB cannot be covered by the range of possible values, but you can still choose to use these units to get fractional display ("0.0035 ZB" for instance).
     */
-    public var allowedUnits: NSByteCountFormatterUnits
+    internal var _allowedUnits: NSByteCountFormatterUnits = .UseDefault
```

Comment:
technically this pattern is not needed and you could write it as the simpler version like:
```swift
public var allowedUnits: NSByteCountFormatterUnits = .UseDefault
```